src/dispatch/task/service.py

```diff
@@ -57,42 +58,86 @@ def get_overdue_tasks(*, db_session) -> List[Optional[Task]]:
     )
 
 
-def create(
-    *,
-    db_session,
-    creator: Participant,
-    assignees: List[Participant],
-    description: str,
-    status: TaskStatus,
-    tickets: List[Ticket],
-    resource_id: str,
-    resource_type: str,
-    weblink: str,
-) -> Task:
+def create(*, db_session, task_in: TaskCreate, creator_email: str) -> Task:
     """Create a new task."""
+    incident = incident_service.get(db_session=db_session, incident_id=task_in.incident.id)
+
+    tickets = [
+        ticket_service.get_or_create_by_weblink(db_session=db_session, weblink=t.weblink)
+        for t in task_in.tickets
+    ]
+
+    assignees = []
+    for i in task_in.assignees:
+        assignees.append(
+            incident_flows.incident_add_or_reactivate_participant_flow(
+                db_session=db_session, incident_id=incident.id, user_email=i.individual.email,
+            )
+        )
+
+    # add creator as a participant if they are not one already
+    creator = incident_flows.incident_add_or_reactivate_participant_flow(
+        db_session=db_session, incident_id=incident.id, user_email=creator_email,
+    )
+
+    # we add owner as a participant if they are not one already
+    if task_in.owner:
+        owner = incident_flows.incident_add_or_reactivate_participant_flow(
+            db_session=db_session,
+            incident_id=incident.id,
+            user_email=task_in.owner.individual.email,
+        )
+    else:
+        owner = incident.commander
+
     task = Task(
+        **task_in.dict(exclude={"assignees", "owner", "incident", "creator", "tickets"}),
         creator=creator,
+        owner=owner,
         assignees=assignees,
-        description=description,
-        status=status,
+        incident=incident,
         tickets=tickets,
-        resource_id=resource_id,
-        resource_type=resource_type,
-        weblink=weblink,
     )
+
+    event_service.log(
+        db_session=db_session,
+        source="Dispatch Core App",
+        description="New task created",
+        incident_id=incident.id,
+    )
```

Comment:
You could use the `details` column in the event model to add the description of the task or any other details, which then could be displayed in the timeline view. For example:
```
details={"description": description},
```

Reply:
That feels like duplicating data on the event timeline (the description could be quite big). If we felt that data was useful for the timeline I would rather merge it instead of duplicating actual data (what happens if a task changes?), at most I would add a link to the task in question here.

Reply:
A link to the task would be great. I wasn't sure we could generate them w/o adding more code.

Reply:
Added.

Reply:
Hmm where did you add it? I don't see it.